build_projects/dotnet-cli-build/TarGzFileCreateFromDirectory.cs

```diff
@@ -0,0 +1,143 @@
+
+        /// <summary>
+        /// If zipping an entire folder without exclusion patterns, whether to include the folder in the archive.
+        /// </summary>
+        public bool IncludeBaseDirectory { get; set; }
+
+        /// <summary>
+        /// An item group of regular expressions for content to exclude from the archive.
+        /// </summary>
+        public ITaskItem[] ExcludePatterns { get; set; }
+
+        protected override bool ValidateParameters()
+        {
+            base.ValidateParameters();
+
+            var retVal = true;
+
+            if (File.Exists(DestinationArchive))
+            {
+                if (OverwriteDestination == true)
+                {
+                    Log.LogMessage(MessageImportance.Low, $"{DestinationArchive} will be overwritten");
+                }
+                else
+                {
+                    Log.LogError($"'{DestinationArchive}' already exists. Did you forget to set '{nameof(OverwriteDestination)}' to true?");
+
+                    retVal = false;
+                }
+            }
+
+            SourceDirectory = Path.GetFullPath(SourceDirectory);
+
+            SourceDirectory = SourceDirectory.EndsWith(Path.DirectorySeparatorChar.ToString())
+                ? SourceDirectory
+                : SourceDirectory + Path.DirectorySeparatorChar;
+
+            if (!Directory.Exists(SourceDirectory))
+            {
+                Log.LogError($"SourceDirectory '{SourceDirectory} does not exist.");
+
+                retVal = false;
+            }
+
+            return retVal;
+        }
+
+        public override bool Execute()
+        {
+            return base.Execute();
+        }
+
+        protected override string ToolName
+        {
+            get { return "tar"; }
+        }
+
+        protected override MessageImportance StandardOutputLoggingImportance
+        {
+            get { return MessageImportance.High; } // or else the output doesn't get logged by default
+        }
+
+        protected override string GenerateFullPathToTool()
+        {
+            return "tar";
+        }
+
+        protected override string GenerateCommandLineCommands()
+        {
+            return $"{GetDestinationArchive()} {GetSourceSpecification()}";
+        }
+
+        private string GetSourceSpecification()
+        {
+            if (IncludeBaseDirectory)
+            {
+                var parentDirectory = Directory.GetParent(SourceDirectory).Parent.FullName;
+
+                var sourceDirectoryName = Path.GetFileName(Path.GetDirectoryName(SourceDirectory));
+
+                return $"--directory {parentDirectory} {sourceDirectoryName}  {GetExcludes()}";
+            }
+            else
+            {
+                return $"--directory {SourceDirectory}  {GetExcludes()} \".\"";
+            }
+        }
+
+        private string GetDestinationArchive()
+        {
+            return $"-czf {DestinationArchive}";
```

Comment:
yeah that's better I think, just wanted to call it out because the extension wasn't getting set at all when I made the comment